app/javascript/react/screens/App/Overview/screens/MappingWizard/components/MappingWizardDatastoresStep/MappingWizardDatastoresStep.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Field, reduxForm } from 'redux-form';
+import { noop, bindMethods } from 'patternfly-react';
+import SourceClusterSelect from './components/SourceClusterSelect/SourceClusterSelect';
+import DatastoresStepForm from './components/DatastoresStepForm';
+
+class MappingWizardDatastoresStep extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedCluster: undefined,
+      selectedClusterMapping: null
+    };
+
+    bindMethods(this, ['selectSourceCluster', 'resetState']);
+  }
+
+  componentDidMount() {}
+
+  selectSourceCluster(sourceClusterId) {
+    // when dropdown selection occurs for source cluster, we go retrieve the datastores for that
+    // cluster
+    const {
+      fetchDatastoresUrl,
+      fetchSourceDatastoresAction,
+      fetchTargetDatastoresAction,
+      clusterMappings
+    } = this.props;
+
+    const selectedClusterMapping = clusterMappings.find(clusterMapping =>
+      clusterMapping.nodes.some(
+        sourceCluster => sourceCluster.id === sourceClusterId
+      )
+    );
+
+    const { nodes: sourceClusters, ...targetCluster } = selectedClusterMapping;
+
+    this.setState(() => ({
+      selectedCluster: sourceClusters.find(
+        sourceCluster => sourceCluster.id === sourceClusterId
+      ),
+      selectedClusterMapping
+    }));
+
+    fetchSourceDatastoresAction(fetchDatastoresUrl, sourceClusterId);
+    fetchTargetDatastoresAction(fetchDatastoresUrl, targetCluster.id);
+  }
+
+  resetState() {
+    this.setState(() => ({
+      selectedCluster: undefined,
+      selectedClusterMapping: null
+    }));
+  }
+
+  render() {
+    const {
+      clusterMappings,
+      isFetchingSourceDatastores,
+      isRejectedSourceDatastores, // eslint-disable-line no-unused-vars
+      isFetchingTargetDatastores,
+      isRejectedTargetDatastores, // eslint-disable-line no-unused-vars
+      // source/target datastores change depending on selection
+      sourceDatastores,
+      targetDatastores,
+      removeSourceDatastores,
+      removeTargetDatastore,
+      addSourceDatastores,
+      addTargetDatastore
+    } = this.props;
+
+    const { selectedCluster, selectedClusterMapping } = this.state;
+
+    // first we render the dropdown selection for each source cluster in clusterMappings,
+    // then we call `selectSourceCluster` and go get that cluster's datastores on selection
+    return (
+      <div>
+        <SourceClusterSelect
```

Comment:
i agree that this state should be lifted so that we can use it to validate next button state. I think that can be addressed in a future PR though. Again - would like to get the base function of adding mappings and connected to the APIs.